lib/javascript/JavascriptParser.js

```diff
@@ -3162,9 +3163,10 @@ class JavascriptParser extends Parser {
 	}
 
 	getComments(range) {
-		return this.comments.filter(
-			comment => comment.range[0] >= range[0] && comment.range[1] <= range[1]
-		);
+		const [rangeStart, rangeEnd] = range;
+		const startIdx = binarySearch(this.comments, i => i.range[0] >= rangeStart);
+		const endIdx = binarySearch(this.comments, i => i.range[1] > rangeEnd);
```

Comment:
> Instead of searching the end, just step forward from the start until the end. While going forward add items to the result array.

https://github.com/webpack/webpack/commit/30e36b274a16e521ce79f02952ee7e2e78b9c012